### ISSUE TITLE
Error compiling previously working project - NPE inside groovy compiler

### ISSUE THREAD

**Issue by ssadedin** [groovy/groovy-eclipse#1432]
I have a fairly large codebase that was working previously but now is failing to build with the latest groovy-eclipse.

When Eclipse tries to build the project, the following error is shown:

![image](https://user-images.githubusercontent.com/138868/210553645-2d9fe9e5-c692-4804-8993-e43b0df02819.png)

The error trace from the eclipse log is:

```
java.lang.NullPointerException: Cannot read the array length because "this.referenceContext.fields" is null
	at org.codehaus.jdt.groovy.internal.compiler.ast.GroovyClassScope.buildFieldsAndMethods(GroovyClassScope.java:418)
	at org.eclipse.jdt.internal.compiler.lookup.CompilationUnitScope.buildFieldsAndMethods(CompilationUnitScope.java:105)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.completeTypeBindings(LookupEnvironment.java:537)
	at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:894)
	at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:404)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:454)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:436)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:402)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:218)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:334)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:80)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:273)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:193)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:1020)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:247)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:392)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:395)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:506)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:454)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:536)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:161)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:255)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

I am hoping this is enough context to investigate the problem. If not, I can try to narrow down what part of the code is causing this. I have other projects within the same workspace that do not experience this issue and build correctly. Hence I believe it's likely something to do with the specific code in one of the classes of the project.

**Comment by eric-milles** [comment]
@ssadedin The NPE is easy to fix.  I'm concerned that it will just move the problem, since I cannot recreate to check out what might happen next.  I don't have any tests that go this path.  The expectation seems to be that a `GroovyTypeDeclaration` always has non-null `fields` array.  Do you have other errors in your _Error Log_ view that might help determine the source of the problem?

**Comment by ssadedin** [comment]
Nothing else in the error log or stderr/stdout from eclipse that looks relevant unfortunately. I will try to narrow this down some more and post some more details. Thanks for checking in!

**Comment by eric-milles** [comment]
You can open the _Groovy Event Console_ from with the _Console_ view and then do a rebuild.  There should be some trace logging for various steps of compilation.

**Comment by ssadedin** [comment]
I went through the exercise of progressively deleting source files until the error went away and then slowly adding them back again. At the end of this I had my complete project back and the error couldn't be reproduced any more. I guess I have to write this off as some kind of bad state in my workspace or project that got reset by what I did.

Hopefully posting this documents it in case someone else runs into the same thing, but for now I will close this as I don't think I can provide you a reproducible example.

Thanks again for the followup and sorry for the noise!
